### PR TITLE
(bunq/sdk_java#55) add response id to request error

### PR DESCRIPTION
**Pull request by OGKevin**
Closes bunq/sdk_java#55
 - [x] Tested

### CODE REVIEW

File: src/test/java/com/bunq/sdk/http/ErrorResponseTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import com.bunq.sdk.BunqSdkTestBase;
+import com.bunq.sdk.exception.ApiException;
+import com.bunq.sdk.model.generated.endpoint.UserPerson;
+import org.junit.Test;
+
+import static org.junit.Assert.assertNotNull;
+
+public class ErrorResponseTest extends BunqSdkTestBase {
+  /**
```
Comment (OGKevin) [inline src/test/java/com/bunq/sdk/http/ErrorResponseTest.java]: Enter

File: src/test/java/com/bunq/sdk/http/ErrorResponseTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    try {
+      UserPerson.get(getApiContext(), INVALID_USER_PERSON_ID);
+    } catch (ApiException $exception) {
+      caughtException = $exception;
+    }
+
+    assertNotNull(caughtException);
+    assertNotNull(caughtException.getResponseId());
+  }
+}
```
Comment (OGKevin) [inline src/test/java/com/bunq/sdk/http/ErrorResponseTest.java]: Enter

File: src/main/java/com/bunq/sdk/exception/ExceptionFactory.java [hunk 3]
```diff
@@ -27,31 +32,31 @@
...
+        return new PleaseContactBunqException(error_message, responseCode, responseId);
       default:
-        return new UnknownApiErrorException(error_message, responseCode);
+        return new UnknownApiErrorException(error_message, responseCode, responseId);
     }
   }
 
-  private static String concatenateMessages(List<String> messages) {
-    return String.join(GLUE_ERROR_MESSAGES, messages);
+  private static String concatenateMessages(List<String> messages, String responseId) {
```
Comment (OGKevin) [inline src/main/java/com/bunq/sdk/exception/ExceptionFactory.java]: Plural 🙅‍♂️

**Comment by OGKevin** [comment]
@patrickdw1991 all yours please 👀

**Comment by OGKevin** [comment]
@andrederoos all yours please 👀

File: src/main/java/com/bunq/sdk/http/ApiClient.java [hunk 4]
```diff
@@ -229,47 +237,59 @@ private BunqResponseRaw createBunqResponseRaw(CloseableHttpResponse response)
...
-      throw createApiExceptionRequestUnsuccessful(responseCode, new String(responseBodyBytes));
+      throw createApiExceptionRequestUnsuccessful(responseCode, new String(responseBodyBytes), responseId);
     }
   }
 
   private static ApiException createApiExceptionRequestUnsuccessful(Integer responseCode,
-      String responseBody) {
-    List<String> errorDescriptions = new ArrayList<>();
+      String responseBody, String responseId) {
+    List<String> AllErrorDescription = new ArrayList<>();
```
Comment (andrederoos) [inline src/main/java/com/bunq/sdk/http/ApiClient.java]: why the capital?

File: src/main/java/com/bunq/sdk/http/ApiClient.java [hunk 5]
```diff
@@ -229,47 +237,59 @@ private BunqResponseRaw createBunqResponseRaw(CloseableHttpResponse response)
...
 
-  private static void assertResponseSuccess(int responseCode, byte[] responseBodyBytes) {
+  private static String getResponseId(CloseableHttpResponse response) {
+    Map<String, String> headerMap = getHeadersMap(response);
+
+    if (headerMap.containsKey(HEADER_RESPONSE_ID_LOWER_CASE)) {
+      return headerMap.get(HEADER_RESPONSE_ID_LOWER_CASE);
+    } else if (headerMap.containsKey(HEADER_RESPONSE_ID_UPPER_CASE)) {
+      return  headerMap.get(HEADER_RESPONSE_ID_UPPER_CASE);
+    }
```
Comment (andrederoos) [inline src/main/java/com/bunq/sdk/http/ApiClient.java]: missing the else...

**Pull request merged**